data/tribes/buildings/productionsites/frisians/charcoal_burners_house/init.lua

```diff
@@ -80,16 +80,16 @@ tribes:new_productionsite_type {
          actions = {
             "return=skipped unless economy needs coal",
             "return=failed unless site has log:3",
```

Comment:
No need to fail here, since the worker won't actually do anything until the consume statement is reached

Reply:
basically there is no difference in skipping and failing anymore. However if no logs available I'd rate this a fail. anyhow this is only to exclude the search from having a positive on statistics.

Reply:
With this implementation, the program will fail immediately if there are no logs. If there are logs but no dry ponds, the program will then fail.

If you remove the "return=failed unless site has log:3", then the program will first check for ponds availability, and if there are no ponds it fails. If there are ponds, "consume" will fail unless there are enough logs.

In either case, the consumption of logs and placing of a stack are carried out only after both checks have succeeded.
So the behaviour would be exactly the same but with one statement less needed